configargparse.py

```diff
@@ -119,6 +121,44 @@ def serialize(self, items):
         """
         raise NotImplementedError("serialize(..) not implemented")
 
+    def _tweak_value(self, key, value, filename):
+        """Ensure that all tweaked values really are strings, or list of strings,
+           or none, in order to update the collection of values.
+        """
+        # Call the function designed to be overridden in subclass.
+        tweaked_dict = self.tweak_value(key, value, filename)
+
+        # Seems a little weird, but anything that is not a list is converted to string,
+        # It will be converted back to boolean, int or whatever after.
+        # Because config values are still passed to argparser for computation.
+        for newkey, newvalue in tweaked_dict.items():
+            if newvalue is None:
+                pass
+            elif isinstance(newvalue, list):
+                # We allow lists of strings, or, for compatibility with action="append",
+                # lists of lists of strings.
+                if all(isinstance(v, list) for v in newvalue):
+                    yield newkey, [[str(v) for v in sublist] for sublist in newvalue]
+                else:
+                    yield newkey, [str(v) for v in newvalue]
+            else:
+                yield newkey, str(newvalue)
+
+    def tweak_value(self, key, value, filename):
```

Comment:
function name: To me "tweak" implies small modification, but the example subclasses use this function in various other ways. I'd vote for changing this to something like `process_value` or `handle_config_entry` (since it takes both the key and the value). 
Also, the docs nicely explain what the return value can contain, but should also clarify what the filename arg is, and that extra key/values in the dict will be treated as if they were found in that config file (?) while keys with a None value will be treated as if they were absent from the config file(?)

Reply:
Agreed. Maybe `amend_config_entry` is a fitting name?

Reply:
how about `process_config_entry` since this method may be useful for scenarios that don't modify the entry?